dask/dataframe/core.py

```diff
@@ -5108,6 +5108,10 @@ def merge(
         3. Joining both on columns. In this case a hash join is performed using
            ``dask.dataframe.multi.hash_join``.
 
+        While joining on indices or columns, make sure every row with same index/column
```

Comment:
Thanks @bryanwweber for discussing this! The memory error happens only on hash joins, so it'll be nicer to phrase this as an FAQ rather than a warning.